docs/developer-guide/debugging.md

```diff
@@ -0,0 +1,164 @@
+# Run WVA locally for debugging (connects to Cluster & Prometheus)
+
+This guide shows how to run the Workload Variant Autoscaler (WVA) locally while letting it communicate with your Kubernetes/OpenShift cluster and Prometheus using an SSH tunnel and a ServiceMonitor.
+
+### Quick summary
+- Purpose: Run WVA locally (IDE or terminal) and let Prometheus in-cluster scrape the local /metrics endpoint via an SSH tunnel.
+- Outcome: Prometheus scrapes your local WVA instance's metrics and WVA can query Prometheus in the cluster.
+
+### Prerequisites
+- kubectl configured to talk to the target cluster (set KUBECONFIG if needed)
+- A local shell that can run ssh and kubectl
+- WVA is installed but scaled down to zero replicas in the cluster
+
+### Steps
+
+1) Create a namespace `debugging` and a secret `ssh-key`:
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ssh-key
+  namespace: debugging
```

Comment:
Do you know if we need to create this in a newspace? In an admin-managed cluster, users are on-boarded into a namespace. Will the user need to ask the admin to create a new debug namespace?